compiler/rustc_passes/src/liveness.rs

```diff
@@ -332,6 +332,13 @@ impl<'tcx> Visitor<'tcx> for IrMaps<'tcx> {
             }
         }
 
+        // Don't run unused pass for #[naked]
+        if self.tcx.has_attr(def_id, sym::naked) {
+            if let DefKind::Fn | DefKind::AssocFn = self.tcx.def_kind(def_id.expect_local()) {
```

Comment:
Why is this line needed? Surely we already have a check somewhere else that `#[naked]` can only be applied to functions (and not closures for example).

Reply:
Great question. It looks like we do have a check [here](https://github.com/rust-lang/rust/blob/master/compiler/rustc_passes/src/check_attr.rs#L214-L239). I'll remove this.